.ci-config/rippled.cfg

```diff
@@ -105,6 +105,10 @@ validators.txt
 # Note: The version of rippled you use this config with must have an implementation for the amendments you attempt to enable or it will crash.
 # If you need the version of rippled to be more up to date, you may need to make a comment on this repo: https://github.com/WietseWind/docker-rippled
 
+# network_id is required otherwise it's read as None
+[network_id]
+63456
```

Comment:
Why do we need a network ID?

Reply:
This is related to the integration test: `test_autofill_populate_networkid`. Originally, this test was using the hooks-testnet for sending transactions. But it was decided that client libraries are not supporting (or) using hooks test-nets.

The docker container for the integration test uses this config file for startup.


The autofill method's [control flow](https://github.com/XRPLF/xrpl-py/blob/70ed9c034ad32aed1c105e042690c420a00a091e/xrpl/asyncio/transaction/main.py#L235)  is contingent on `network_id > 1025` for this particular test.

This change is not directly related to the API v2 updates

Reply:
> But it was decided that client libraries are not supporting (or) using hooks test-nets.

We can still use it for a test.

> This change is not directly related to the API v2 updates

Why is this change included in this PR, then?

Reply:
i believe the integration tests are better off without the dependency on hooks-test-network -- this particular function can be tested without making network calls to the hooks-network

I can remove this change from this PR

Reply:
It'll be easier if that discussion can happen separately.

Reply:
I have reverted the changes to the `main` branch. But it appears that the hooks testnet is using `rippledv1.10` : https://github.com/XRPL-Labs/xrpld-hooks/blob/b13e7db18d1376d40cdbeeea849702b568324279/src/ripple/protocol/impl/BuildInfo.cpp#L36

Here is the Request and Response from hooks testnet validator
```
# request
ServerInfo(method=<RequestMethod.SERVER_INFO: 'server_info'>, id=None, api_version=2)

# response
Response(status=<ResponseStatus.ERROR: 'error'>, result={'api_version': 2, 'error': 'invalid_API_version', 'id': 'RequestMethod.SERVER_INFO_174181', 'request': {'api_version': 2, 'command': 'server_info', 'id': 'RequestMethod.SERVER_INFO_174181'}, 'status': 'error', 'type': 'response'}, id='RequestMethod.SERVER_INFO_174181', type=<ResponseType.RESPONSE: 'response'>)

```

Reply:
I'll try to find a fix tomorrow

Reply:
Make a quick PR to fix those issues first?

Reply:
For the validators using v1 API, we will need to specify `api_version: 1` in *all* the requests. These requests are not localized to a particular file (or) a class.

I need to set a global switch (for `Request.api_version`) based on the API version used by the client. Let me try to see this works

Reply:
I don't think that's necessary, and I'm not really sure how that'd work anyways. People using an older version of rippled that doesn't support APIv2 likely don't need to upgrade to a newer version of the library anyways.

Reply:
> I didn't understand your second sentence, you're saying that there will always be legacy networks (pre `APIv2`), but how do we deal with them?

Right, but they don't benefit from upgrading to newer versions of the libraries. They can continue to use older versions.

Reply:
I see -- can we use another testnet in their stead? We will need a candidate with `network_id > 1024` (Now, I'm able to place your offline question in light of this discussion 👍 )

Reply:
alternatively, we could revert to specifying a network_id inside the config file for the integration tests

Reply:
We need to test two things:
* network ID > 1024
* network ID <= 1024

Regardless of what the standalone config says, we'll only be testing one of those.

Reply:
Why do we need to test `networkID <= 1024` ? The test expects a `RippledError` object stating that `NetworkID` is not required. But should xrpl-py validate all possible RippledErrors? I don't understand the separation of responsibilities between rippled and client libraries.

But, if we need to test both these cases through xrpl-py, then:
1. We need to set up test networks with network_id > 1024 -- This is useful for testing the interfaces of other client libraries too.

2. Use two docker containers with two different network_id's -- this covers both the possibilities. I need to figure out a clean port-mapping of the docker containers to disambiguate the two cases

Reply:
`networkID <= 1024` _must not_ have the`NetworkID` included in the transaction. `networkID > 1024` _must_ have the `NetworkID` included in the transaction. We need to ensure that xrpl-py behaves accordingly. This was what we had the hooks testnet testcase for - to ensure the latter case.

Neither of the scenarios you suggest are really worth the effort just to test one edgecase. The better option is probably to change the standalone config to have a network ID > 1024 and then add a network ID test for one of the testnets that have a network ID < 1024.

Reply:
okay, I've reverted back one of my older commits, and I also added a new test case
[8632aef](https://github.com/XRPLF/xrpl-py/pull/720/commits/8632aef2f8fb981353d5f5ce2df187a415130f61)